config/cesm/machines/config_machines.xml

```diff
@@ -1719,24 +1719,82 @@
+    <MPILIBS>impi,mvapich2</MPILIBS>
     <CIME_OUTPUT_ROOT>$ENV{SCRATCH}</CIME_OUTPUT_ROOT>
     <DIN_LOC_ROOT>/scratch/projects/xsede/CESM/inputdata</DIN_LOC_ROOT>
-    <DIN_LOC_ROOT_CLMFORC>/scratch/projects/xsede/CESM/inputdata/lmwg</DIN_LOC_ROOT_CLMFORC>
+    <DIN_LOC_ROOT_CLMFORC>/scratch/projects/xsede/CESM/inputdata/inputdata/lmwg</DIN_LOC_ROOT_CLMFORC>
     <DOUT_S_ROOT>$WORK/archive/$CASE</DOUT_S_ROOT>
-    <BASELINE_ROOT>/scratch/projects/xsede/CESM/ccsm_baselines</BASELINE_ROOT>
-    <CCSM_CPRNC>/scratch/projects/xsede/CESM/tools/cprnc/cprnc</CCSM_CPRNC>
+    <BASELINE_ROOT>/scratch/projects/xsede/CESM//cesm_baselines</BASELINE_ROOT>
+    <CCSM_CPRNC>/scratch/projects/xsede/CESM/cime/tools/cprnc/cprnc</CCSM_CPRNC>
     <GMAKE_J>4</GMAKE_J>
     <BATCH_SYSTEM>slurm</BATCH_SYSTEM>
     <SUPPORTED_BY>cseg</SUPPORTED_BY>
-    <MAX_TASKS_PER_NODE>16</MAX_TASKS_PER_NODE>
-    <MAX_MPITASKS_PER_NODE>16</MAX_MPITASKS_PER_NODE>
+    <MAX_TASKS_PER_NODE>96</MAX_TASKS_PER_NODE>
+    <MAX_MPITASKS_PER_NODE>48</MAX_MPITASKS_PER_NODE>
+    <mpirun mpilib="impi">
+      <executable>ibrun</executable>
+    </mpirun>
+    <mpirun mpilib="mvapich2">
+      <executable>ibrun</executable>
+    </mpirun>
+    <module_system type="module">
+      <init_path lang="perl">/opt/apps/lmod/lmod/init/perl</init_path>
+      <init_path lang="python">/opt/apps/lmod/lmod/init/env_modules_python.py</init_path>
+      <init_path lang="sh">/opt/apps/lmod/lmod/init/sh</init_path>
+      <init_path lang="csh">/opt/apps/lmod/lmod/init/csh</init_path>
+      <cmd_path lang="perl">/opt/apps/lmod/lmod/libexec/lmod perl</cmd_path>
+      <cmd_path lang="python">/opt/apps/lmod/lmod/libexec/lmod python</cmd_path>
+      <cmd_path lang="sh">module</cmd_path>
+      <cmd_path lang="csh">module</cmd_path>
+      <modules>
+        <command name="purge"></command>
+        <command name="load">TACC</command>
+        <command name="load">python/2.7.13</command>
+        <command name="load">intel/17.0.4</command>
+        <command name="load">cmake/3.10.2</command>
+      </modules>
+      <modules mpilib="mvapich2">
+        <command name="load">mvapich2/2.3b</command>
+        <command name="load">pnetcdf/1.8.1</command>
+        <command name="load">parallel-netcdf/4.3.3.1</command>
+      </modules>
+      <modules mpilib="impi">
+        <command name="rm">mvapich2</command>
+        <command name="load">impi/17.0.3</command>
+        <command name="load">pnetcdf/1.8.1</command>
+        <command name="load">parallel-netcdf/4.3.3.1</command>
+      </modules>
+      <modules mpilib="mpi-serial">
+        <command name="load">netcdf/4.3.3.1</command>
+      </modules>
+    </module_system>
+    <environment_variables>
+      <env name="OMP_STACKSIZE">256M</env>
+     </environment_variables>
+  </machine>
 
+  <machine MACH="stampede2-knl">
+    <DESC>Intel Xeon Phi 7250 ("Knights Landing") , batch system is SLURM</DESC>
+    <NODENAME_REGEX>.*stampede2</NODENAME_REGEX>
```

Comment:
can't have the same NODENAME_REGEX for both definitions - I would suggest removing this one which would make skylake the default and require a --machine argument for knl

Reply:
Not a problem.  I've removed it.